packages/assets-controllers/src/DeFiPositionsController/DeFiPositionsController.ts

```diff
@@ -0,0 +1,245 @@
+      this.startPolling(null);
+    });
+
+    this.messagingSystem.subscribe('KeyringController:lock', () => {
+      this.stopAllPolling();
+    });
+
+    this.messagingSystem.subscribe(
+      'TransactionController:transactionConfirmed',
+      async (transactionMeta) => {
+        if (!this.#isEnabled()) {
+          return;
+        }
+
+        await this.#updateAccountPositions(transactionMeta.txParams.from);
+      },
+    );
+
+    this.messagingSystem.subscribe(
+      'AccountsController:accountAdded',
+      async (account) => {
+        if (!this.#isEnabled() || !account.type.startsWith('eip155:')) {
+          return;
+        }
+
+        await this.#updateAccountPositions(account.address);
+      },
+    );
+  }
+
+  async _executePoll(): Promise<void> {
+    if (!this.#isEnabled()) {
+      return;
+    }
+
+    const accounts = this.messagingSystem.call(
+      'AccountsController:listAccounts',
+    );
+
+    const results = await reduceInBatchesSerially({
+      initialResult: [] as {
+        accountAddress: string;
+        positions: GroupedPositionsPerChain | null;
+      }[],
+      values: accounts,
+      batchSize: FETCH_POSITIONS_BATCH_SIZE,
+      eachBatch: async (workingResult, batch) => {
+        const batchResults = (
+          await Promise.all(
+            batch.map(async ({ address: accountAddress, type }) => {
+              if (type.startsWith('eip155:')) {
+                const positions =
+                  await this.#fetchAccountPositions(accountAddress);
+
+                return {
+                  accountAddress,
+                  positions,
+                };
+              }
+
+              return undefined;
+            }),
+          )
+        ).filter(Boolean) as {
+          accountAddress: string;
+          positions: GroupedPositionsPerChain | null;
+        }[];
+
+        return [...workingResult, ...batchResults];
+      },
+    });
+
+    const allDefiPositions = results.reduce(
+      (acc, { accountAddress, positions }) => {
+        acc[accountAddress] = positions;
+        return acc;
+      },
+      {} as DeFiPositionsControllerState['allDeFiPositions'],
+    );
+
+    this.update((state) => {
+      state.allDeFiPositions = allDefiPositions;
+    });
+  }
+
+  async #updateAccountPositions(accountAddress: string): Promise<void> {
+    const accountPositionsPerChain =
+      await this.#fetchAccountPositions(accountAddress);
+
+    this.update((state) => {
+      state.allDeFiPositions[accountAddress] = accountPositionsPerChain;
+    });
+  }
+
+  async #fetchAccountPositions(
+    accountAddress: string,
+  ): Promise<GroupedPositionsPerChain | null> {
+    try {
+      const defiPositionsResponse = await this.#fetchPositions(accountAddress);
+
+      return groupPositions(defiPositionsResponse);
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    } catch (error) {
+      return null;
```

Comment:
This is how we are going to represent that an error has occurred.

Reply:
Nit: maybe we can remove the // eslint-disable-next-line @typescript-eslint/no-unused-vars
and just
`catch { return null }`